### DISCUSSION TITLE
how can I transfer data from the parent data loader to the child data loader

### DISCUSSION THREAD

**Discussion by aymenbraiek** [Netflix/dgs-framework#1711] [General]
how can I implements such methods ?
public CompletableFuture<DataFetcherResult<List<Show>>> shows(...) {...
public DataFetcherResult<CompletableFuture<List<Show>>> shows(...) {...

**Comment by GregLyons** [comment]
Let me know if I'm misunderstanding the question, but I'm assuming you have a parent field and a child field, both that use data loaders, and you want to use the result of the parent field's `DataLoader` in the child field's dataloader:

```gql
type Parent {
    fieldOnParent: Child
}

type Child {
    fieldOnChild: String
}
```

In this case, I interpret your question as saying you're in the data fetcher for `Child#fieldonChild`, which uses a `DataLoader`, and you want to pass the result of the `Parent#fieldOnParent` data fetcher (which also uses a `DataLoader`) in the `DataLoader` for `fieldOnChild`. In this case, the result of `Parent#fieldOnParent` is a `Child`, and you can pass it down to the `DataLoader` for `Child#fieldOnChild` by having the `DataLoader` for `Child#fieldOnChild` implement `MappedBatchLoaderWithContext` and then doing something like:

```java
@DgsData(parentType = DgsConstants.CHILD.TYPE_NAME, field = DgsConstants.CHILD.FieldOnChild)
public CompletableFuture<String> fieldOnChild(DgsDataFetchingEnvironment dfe) {
    DataLoader<String, String> dataLoader = dfe.getDataLoader(ChildDataLoader.class);
    Child source = dfe.getSource();
    // `MappedBatchLoaderWithContext` lets you load in the entire `source` as well
    return dataLoader.load(source.getId(), source);
}
```

Then you can extract the source (mapped by key) using `BatchLoaderEnvironment#getKeyContexts()`. I believe you can use other DGS context using `BatchLoaderEnvironment#getContext` as well, see the bottom of this [Data Fetching Cotnext docs page](https://netflix.github.io/dgs/advanced/custom-datafetcher-context/).

Hope this helps
